kpi/deployment_backends/kc_access/shadow_models.py

```diff
@@ -395,6 +398,30 @@ def set_mfa_status(cls, user_id: int, is_active: bool):
             user_profile.is_mfa_active = int(is_active)
             user_profile.save(update_fields=['is_mfa_active'])
 
+    @classmethod
+    def set_password_details(
+        cls, 
+        user_id: int, 
+        date: datetime, 
+        validated: bool,
+    ):
+        """
+        Update the kobocat user's password_change_date and validated_password fieds
+        :param user_id: the kobocat user's id
+        :param date: the date of the password change
+        :validated: whether the password has been validated or not
+        """
+        try:
+            user_profile, created = cls.objects.get_or_create(user_id=user_id)
+        except cls.DoesNotExist:
+            pass
+        else:
+            user_profile.password_change_date = date
```

Comment:
@jnm said: 

> I'd imagined that KPI would set a flag on the user in KoboCAT that determined whether or not an account should be restricted for any reason, and that KoboCAT wouldn't need to track password change dates

Indeed, KC does not need to track the password change dates. You can remove it from the sync. I'll update [KC PR](https://github.com/kobotoolbox/kobocat/pull/887) too